kratos/geometries/brep_curve_on_surface.h

```diff
@@ -338,6 +338,70 @@ class BrepCurveOnSurface
             mCurveNurbsInterval.GetT0(), mCurveNurbsInterval.GetT1());
     }
 
+    ///@}
+    ///@name IsInside
+    ///@{
+
+    /// returns if rPointLocalCoordinates[0] is inside -> 1 or outside -> 0
+    int IsInsideLocalSpace(
+        const CoordinatesArrayType& rPointLocalCoordinates,
+        const double Tolerance = std::numeric_limits<double>::epsilon()
+    ) const override
+    {
+        const double min_parameter = mCurveNurbsInterval.MinParameter();
+        if (rPointLocalCoordinates[0] < min_parameter) {
+            return 0;
+        }
+        else if (std::abs(rPointLocalCoordinates[0] - min_parameter) < Tolerance) {
+            return 2;
+        }
+
+        const double max_parameter = mCurveNurbsInterval.MaxParameter();
+        if (rPointLocalCoordinates[0] > max_parameter) {
+            return 0;
+        }
+        else if (std::abs(rPointLocalCoordinates[0] - max_parameter) < Tolerance) {
+            return 2;
+        }
+
+        return 1;
+    }
+
+    ///@}
+    ///@name ClosestPoint
+    ///@{
+
+    /* Returns if rPointLocalCoordinates[0] is 
+     *         on boundary -> 2
```

Comment:
Here 2 means something

Reply:
Can you make fancier these commentaries?

Reply:
Yep, I'll do that.